### PR TITLE
Harden mk_lib_database: SQL bug, hardcoded key, panics

### PR DESCRIPTION
**Pull request by SpootDev**
## Summary
Focused hardening pass on `mk_lib_database`:

- **game_servers**: fix `VALUES($ 1, $2, $3)` — the stray space made the statement fail to parse at runtime.
- **library**: make `DBLibraryPathStatus` fields `pub` so the `FromRow` derive and external callers can actually read them.
- **user**: replace `pool.unwrap()` in `load_user` with a proper `Err` so a missing `PgPool` doesn't abort the process.
- **network_share**:
  - Stop embedding the literal `'fake-strong-key'` in every `pgp_sym_decrypt` call. A `share_auth_key()` helper reads `MK_SHARE_AUTH_KEY` from the environment and the value is passed as a bind parameter; the legacy key remains the fallback so unmigrated deployments keep working.
  - Replace the unchecked `path_vec[1]` index with a `parse_share_subpath` helper that returns a descriptive error instead of panicking on inputs that don't contain a share segment.
  - Fix `update_user_info`: it was binding a `String` into a uuid column and ignoring the `password` argument entirely. Signature now takes a `Uuid` and drops the unused password param (password writes belong on `mm_share_auth`).
- **version_schema**: bind the share-auth key in the schema migration too.

Out of scope for this PR: `.unwrap()` calls in the `metadata/*` parsers — those warrant a separate, dedicated pass.

## Test plan
- [ ] `cargo check -p mk_lib_database` in an environment with access to the Kellnr registry.
- [ ] Confirm `MK_SHARE_AUTH_KEY` env var is set in production deployments before rotating; rows written with the legacy key remain decryptable as long as the env var is unset.
- [ ] Exercise share insert/exists paths with a malformed share path to confirm the new `Protocol` error surfaces rather than panicking.
- [ ] Audit callers of `mk_lib_database_network_share_update_user_info` in downstream binaries (none in-workspace) for the signature change.

https://claude.ai/code/session_0156KVpZNzCVtt2TwxxEmr3i

### CODE REVIEW

File: src/mk_lib_database/src/mk_lib_database_network_share.rs [hunk 1]
```diff
@@ -2,19 +2,40 @@ use serde::{Deserialize, Serialize};
...
+/// configurable. Set the env var in production and rotate existing rows.
+fn share_auth_key() -> String {
+    std::env::var("MK_SHARE_AUTH_KEY").unwrap_or_else(|_| "fake-strong-key".to_string())
+}
+
+fn parse_share_subpath(network_share_path: &str) -> Option<String> {
+    let path_name = network_share_path.replace("\\\\", "/");
+    path_name
+        .splitn(3, '/')
+        .nth(1)
```
[the remark this answers is not in the document]
Comment (SpootDev) [reply src/mk_lib_database/src/mk_lib_database_network_share.rs]: Good catch — confirmed. `splitn(3, '/')` + `.nth(1)` returns `""` for `//host/share/...` and includes the host for `\\host\share`.

Fixed in d40499fb: renamed to `parse_share_name`, which normalises backslashes to forward slashes, strips the UNC host prefix, and returns only the share segment. Host already lives in `mm_network_share_ip`, so the path column now stores just the share name. Added unit tests for both UNC forms plus edge cases (bare name, missing share segment, empty string).

---
_Generated by [Claude Code](https://claude.ai/code)_
